### PR TITLE
feat(autoscale): add support for autoscale per proc type on an application

### PR DESCRIPTION
**Pull request by helgi**
ref https://github.com/deis/controller/pull/1018

### CODE REVIEW

File: appsettings/appsettings_test.go [hunk 1]
```diff
@@ -29,17 +30,18 @@ const appSettingsUnsetFixture string = `
 {
     "owner": "test",
     "app": "unset-test",
-	  "maintenance": true,
+	"maintenance": true,
```
Comment (Joshua-Anderson) [inline appsettings/appsettings_test.go]: Are there mixed tabs and spaces here?

Comment (helgi) [inline appsettings/appsettings_test.go]: Apparently so - that was the case before so I am sure it "duplicate" when i added extra lines. I can remove real quick if that is supposed to be spaces?

Comment (Joshua-Anderson) [inline appsettings/appsettings_test.go]: I don't really care either way, I would just prefer all spaces or all tabs across a single file :)

File: api/appsettings.go [hunk 2]
```diff
@@ -30,3 +36,28 @@ func NewRoutable() *bool {
 type Whitelist struct {
 	Addresses []string `json:"addresses"`
 }
+
+// String displays the Autoscale rule in a readable format.
+func (a Autoscale) String() string {
```
Comment (Joshua-Anderson) [inline api/appsettings.go]: Could you possibly write a test for this?

Comment (helgi) [inline api/appsettings.go]: Do we have examples of that?

Comment (Joshua-Anderson) [inline api/appsettings.go]: https://github.com/deis/controller-sdk-go/blob/master/api/config_test.go#L8

Comment (helgi) [inline api/appsettings.go]: Added

**Pull request merged**